x-pack/plugins/task_manager/server/task_running/errors.ts

```diff
@@ -52,14 +60,10 @@ export function isRetryableError(error: Error | DecoratedError) {
   return null;
 }
 
-export function createRetryableError(error: Error, shouldRetry: Date | boolean): DecoratedError {
+export function throwRetryableError(error: Error, shouldRetry: Date | boolean) {
```

Comment:
do we want to specify an error source for retryable errors?

Reply:
Yes we are, this PR has 2 follow-on issues, one for actions and one for alerting plugins.
`throwRetryableError` will be handled with actions plugin as it's used only in it.

https://github.com/elastic/kibana/issues/168631
https://github.com/elastic/kibana/issues/168635

Reply:
If there are multiple PRs before the feature is complete, should we consider merging to a feature branch?

Reply:
I don't think that it's needed.
This PR is shippable and doesn't cause any problem or break something.
I'll just add the source data to the reported errors and split the metrics accordingly.